src/librustc_typeck/diagnostics.rs

```diff
@@ -1368,6 +1368,52 @@ struct Foo {
 ```
 "##,
 
+E0128: r##"
+You declared a type parameter with a default which has the same identifier as
+the type parameter. Erroneous code example:
+
+```
+pub struct Foo<SomeType=SomeType>;
+// error: type parameters with a default cannot use forward declared
+// identifiers
+```
+
+Please verify you used the good type or change the type parameter identifier.
+Example:
+
+```
+pub struct Foo<T=SomeType> {
+    value: T
+}
+```
+"##,
+
+E0130: r##"
+You declared a pattern as an argument in a foreign function declaration.
+Erroneous code example:
+
+```
+extern {
+    fn foo((a, b): (u32, u32)); // error: patterns aren't allowed in foreign
+                                //        function declarations
+}
+```
+
+Please remove the pattern argument in the foreign function declaration and
```

Comment:
Not necessary to define  a struct. We're just declaring it, so the pattern can be replaced by a direct binding, e.g. `x: (u32, u32)`

"Please replace the pattern argument with a regular one"